services/ark-api/ark-api/src/ark_api/api/v1/openai.py

```diff
@@ -79,7 +145,25 @@ async def chat_completions(request: ChatCompletionRequest) -> ChatCompletion:
             await ark_client.queries.a_create(query_resource)
             logger.info(f"Created query: {query_name}")
 
-            # Poll for completion using helper function
+            # OpenAI spec: if stream=true, client expects SSE format regardless of downstream support
```

Comment:
This is messy. we should have a function to check if the event stream is available (i.e. check if we have an associated endpoint with a streaming address). If we do, but there is no valid address, we fail - we have a supposed streaming endpoint but it is not resolvable.

Then if streaming is requested and we don't have a streaming backend, we simply poll for completion. otherewise we connect to the streaming endpoint. be much more clear with this flow it is important.

Not having a streaming backend is fine - we poll, having a streaming backend that is misconfigured or errors is an error, don't mask errors.